addons/l10n_ec_website_sale/controllers/main.py

```diff
@@ -0,0 +1,42 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.website_sale.controllers.main import WebsiteSale
+from odoo.http import request
+
+
+class L10nECWebsiteSale(WebsiteSale):
+
+    def _get_mandatory_fields_billing(self, country_id=False):
+        """Extend mandatory fields to add new identification and responsibility fields when company is Ecuador"""
+        res = super()._get_mandatory_fields_billing(country_id)
+        if request.website.sudo().company_id.country_id.code == "EC":
+            res += ["l10n_latam_identification_type_id", "vat"]
+        return res
+
+    def _get_country_related_render_values(self, kw, render_values):
+        res = super()._get_country_related_render_values(kw, render_values)
+        if request.website.sudo().company_id.country_id.code == "EC":
+            res.update({
+                'identification': kw.get('l10n_latam_identification_type_id'),
+                'identification_types': request.env['l10n_latam.identification.type'].search(
+                    ['|', ('country_id', '=', False), ('country_id.code', '=', 'EC')]),
+            })
+        return res
+
+    def _get_vat_validation_fields(self, data):
+        res = super()._get_vat_validation_fields(data)
+        latam_id_type_data = data.get("l10n_latam_identification_type_id")
+        if request.website.sudo().company_id.country_id.code == "EC":
+            res.update({
+                'l10n_latam_identification_type_id': int(latam_id_type_data) if latam_id_type_data else False,
+                'name': data.get('name', False),
+            })
+        return res
+
+    def _get_shop_payment_values(self, order, **kwargs):
+        payment_values = super()._get_shop_payment_values(order, **kwargs)
+        company = order.company_id
+        if company.account_fiscal_country_id.code == 'EC':
+            payment_methods = payment_values['payment_methods_sudo'].filtered(lambda pm: bool(pm.l10n_ec_sri_payment_id))
+            payment_values['payment_methods_sudo'] = payment_methods
```

Comment:
I have some doubts if may this could not be neccesary. The payment method is already passed to the backend, and with that information the sale order throught the method _create_invoices() can create the invoice computing the l10n_ec_sri_payment_id from the payment method.

Reply:
Here we "hide" the payment methods that do not have a `l10n_ec_sri_payment_id` set. They are not available in the shop. This is to avoid the problem you mentioned below. I added a comment in case it was not clear.

I also talked to @jco-odoo I will try to add a value for the SRI Payment Method to some payment methods.
I added "Credit Card" to the "Card" Payment Method for now.
(So that they are available by default and nothing has to be configured for them.)
cc: @vbe-odoo . What do you think?

Reply:
@svfu-odoo 
It may be helpful for the user indeed, to have. default value. Just in case in the case of "Card", both Credit or Debit cards can be used but still seems a good idea to have a default value, like saying that when using Card then it will be a Credit Card. 

For "Wire Transfers", the SRI Payment Method called "Others with use of the financial system" can be used, as it seems the most proper one there.

Those are the only 2 I see that could be mapped... I don't know what you think @pepetreshere ?
cc' @jco-odoo